pandas/tests/indexes/interval/test_interval.py

```diff
@@ -889,6 +889,38 @@ def test_is_all_dates(self):
         year_2017_index = IntervalIndex([year_2017])
         assert not year_2017_index._is_all_dates
 
+    def test_IntervalIndex_get_indexer_non_unique(self):
+        # GH30178
+        int_inx = interval_range(Timestamp("2018-01-02"), freq="3D", periods=3)
+
+        target = IntervalIndex(
+            [
+                Interval(
+                    Timestamp("2018-01-02"),
+                    Timestamp("2018-01-05"),
+                    closed="both",
+                ),
+                Interval(
+                    Timestamp("2018-01-05"),
+                    Timestamp("2018-01-08"),
+                    closed="both",
+                ),
+                Interval(
+                    Timestamp("2018-01-08"),
+                    Timestamp("2018-01-11"),
+                    closed="both",
+                ),
+            ]
+        )
+
+        actual1 = int_inx.get_indexer_non_unique(target)
+        actual2 = int_inx.get_indexer_non_unique(pd.array(target))
+        actual3 = int_inx.get_indexer_non_unique(list(target))
+        expected = (np.array([0, 1, 2], dtype=np.int64), np.array([], dtype=np.int64))
+
+        for actual in [actual1, actual2, actual3]:
```

Comment:
Could you use `pytest.mark.parameterize`?

Reply:
Are there any additional functions to check? Like Series?

Reply:
> Could you use `pytest.mark.parameterize`?

Sure! Should I do it like this or there is a better way to avoid repetition?
````
@pytest.mark.parametrize("target",
    [
        IntervalIndex(
            [
                Interval(
                    Timestamp("2018-01-02"),
                    Timestamp("2018-01-05"),
                    closed="both",
                ),
                Interval(
                    Timestamp("2018-01-05"),
                    Timestamp("2018-01-08"),
                    closed="both",
                ),
                Interval(
                    Timestamp("2018-01-08"),
                    Timestamp("2018-01-11"),
                    closed="both",
                ),
            ]
        ),
        pd.array(
            IntervalIndex(
            [
                Interval(
                    Timestamp("2018-01-02"),
                    Timestamp("2018-01-05"),
                    closed="both",
                ),
                Interval(
                    Timestamp("2018-01-05"),
                    Timestamp("2018-01-08"),
                    closed="both",
                ),
                Interval(
                    Timestamp("2018-01-08"),
                    Timestamp("2018-01-11"),
                    closed="both",
                ),
            ]
        )
        ),
        list(
            IntervalIndex(
            [
                Interval(
                    Timestamp("2018-01-02"),
                    Timestamp("2018-01-05"),
                    closed="both",
                ),
                Interval(
                    Timestamp("2018-01-05"),
                    Timestamp("2018-01-08"),
                    closed="both",
                ),
                Interval(
                    Timestamp("2018-01-08"),
                    Timestamp("2018-01-11"),
                    closed="both",
                ),
            ]
        )
        )
    ])
        
    def test_IntervalIndex_get_indexer_non_unique(self,target):
        # GH30178
        int_inx = interval_range(Timestamp("2018-01-02"), freq="3D", periods=3)

        actual = int_inx.get_indexer_non_unique(target)
        expected = (np.array([0, 1, 2], dtype=np.int64), np.array([], dtype=np.int64))

        tm.assert_equal(actual, expected)
`